packages/cactus-plugin-keychain-azure-kv/src/main/typescript/plugin-keychain-azure-kv.ts

```diff
@@ -0,0 +1,257 @@
+import { Server } from "http";
+import { Server as SecureServer } from "https";
+
+import type { Express } from "express";
+import { Optional } from "typescript-optional";
+
+import {
+  Logger,
+  Checks,
+  LogLevelDesc,
+  LoggerProvider,
+} from "@hyperledger/cactus-common";
+import {
+  ICactusPlugin,
+  ICactusPluginOptions,
+  IPluginWebService,
+  IWebServiceEndpoint,
+  PluginAspect,
+} from "@hyperledger/cactus-core-api";
+
+import { KeyVaultSecret, SecretClient } from "@azure/keyvault-secrets";
+import {
+  UsernamePasswordCredential,
+  DefaultAzureCredential,
+} from "@azure/identity";
+
+import { PluginKeychainAzureKvMock } from "../../test/typescript/mock/plugin-keychain-azure-kv-mock";
+
+// TODO: Writing the getExpressRequestHandler() method for
+// GetKeychainEntryEndpointV1 and SetKeychainEntryEndpointV1
+// import { GetKeychainEntryEndpointV1 } from "./web-services/get-keychain-entry-endpoint-v1";
+// import { SetKeychainEntryEndpointV1 } from "./web-services/set-keychain-entry-endpoint-v1";
+
+export enum AzureCredentialType {
+  LocalFile = "LOCAL_FILE",
+  InMemory = "IN_MEMORY",
+}
+
+export interface IAzureInMemoryCredentials {
+  azureTenantId: string;
+  azureClientId: string;
+  azureUsername: string;
+  azurePassword: string;
+}
+
+export interface IPluginKeychainAzureKvOptions extends ICactusPluginOptions {
+  logLevel?: LogLevelDesc;
+  keychainId: string;
+  instanceId: string;
+  azureEndpoint: string;
+  azureCredentialType: AzureCredentialType;
+  azureInMemoryCredentials?: IAzureInMemoryCredentials;
+}
+
+export class PluginKeychainAzureKv implements ICactusPlugin, IPluginWebService {
+  public static readonly CLASS_NAME = "PluginKeychainAzureKv";
+
+  readonly vaultUrl: string;
+  private readonly log: Logger;
+  private readonly instanceId: string;
+  private endpoints: IWebServiceEndpoint[] | undefined;
+  private azureKvClientMock: PluginKeychainAzureKvMock;
+  private readonly secrets: Map<string, string>;
```

Comment:
@jagpreetsinghsasan Do we need this?

Reply:
Yes, I am storing the secrets in this Map<string,string> structure

Reply:
@jagpreetsinghsasan Then it should be in the mock class instead